backend/app/controllers/spree/admin/stock_transfers_controller.rb

```diff
@@ -19,18 +19,22 @@ def show
       def new; end
 
       def create
-        variants = Hash.new(0)
-        params[:variant].each_with_index do |variant_id, i|
-          variants[variant_id] += params[:quantity][i].to_i
+        if params[:variant].nil?
+          flash[:error] = Spree.t('stock_transfer.errors.must_have_variant')
+          render :new
+        else
+          variants = Hash.new(0)
+          params[:variant].each_with_index do |variant_id, i|
+            variants[variant_id] += params[:quantity][i].to_i
```

Comment:
Shouldn't we also consider rejecting the transfers which have 0 or nil as quantity as I don't find any reason to create a stock transfer with a zero quantity?
@damianlegawiec